source/ast/expressions/AssertionExpr.cpp

```diff
@@ -523,22 +625,44 @@ void SimpleAssertionExpr::requireSequence(const ASTContext& context, DiagCode co
     }
 }
 
-bool SimpleAssertionExpr::admitsEmptyImpl() const {
+bitmask<NondegeneracyStatus> SimpleAssertionExpr::checkNondegeneracyImpl() const {
+    if (expr.kind == ExpressionKind::AssertionInstance) {
```

Comment:
I'm not sure this is correct; this ignores the possible repetition when the sequence is a reference to another sequence. The repetition could be for [*0] meaning it would admit only empty matches, regardless of the instance's nondegeneracy status.

Reply:
fixed